poleno/utils/static/poleno/js/collapsable.js

```diff
@@ -0,0 +1,61 @@
+/* Enables elements to automatically show or hide other elements by bootstrap collapsing.
+ * Works on all elements type, which must contain data-toggle="collapse" and
+ * data-target="`css-selector`" attributes. Collapsable element must contain class="collapse".
+ *
+ * data-target-shown elements are shown, when collapsable elements are hidden. data-target-hidden
+ * elements are shown, when collapsable elements are shown and vice versa.
+ *
+ * By entering url with a hash into browser, content and elements will be automatically collapsed.
+ *
+ * Requires:
+ *  -- JQuery
+ *
+ * Examples:
+ *  <div class="parent">
+ *    <div data-toggle="collapse" data-target="#content">
+ *      <p class="collapse-shown">Collapse</p>
+ *      <p class="is-hidden collapse-hidden">Collapsed</p>
+ *    </div>
+ *    <p id="content" class="collapse" data-target-shown=".collapse-shown"
+ *       data-target-hidden=".collapse-hidden" data-container=".question">
+ *      Content
+ *    </p>
+ *  </div>
+ */
+$(function(){
+	function collapse_show(){
+		var container = $(this).data('container') || 'html';
+		var target_shown = $.map(this.attributes, function(attr){
+			if (attr.name.match("^data-target-shown")) return attr.value;
+		}).join(', ');
+		var target_hidden = $.map(this.attributes, function(attr){
+			if (attr.name.match("^data-target-hidden")) return attr.value;
+		}).join(', ');
+		$(this).closest(container).find(target_shown).hide();
+		$(this).closest(container).find(target_hidden).show();
+	}
+	function collapse_hide(){
+		var container = $(this).data('container') || 'html';
+		var target_shown = $.map(this.attributes, function(attr){
+			if (attr.name.match("^data-target-shown")) return attr.value;
+		}).join(', ');
+		var target_hidden = $.map(this.attributes, function(attr){
+			if (attr.name.match("^data-target-hidden")) return attr.value;
+		}).join(', ');
+		$(this).closest(container).find(target_shown).show();
+		$(this).closest(container).find(target_hidden).hide();
+	}
+	$('.collapse').on('show.bs.collapse', collapse_show);
+	$('.collapse').on('hide.bs.collapse', collapse_hide);
+	$('[data-toggle="collapse"]').on('click', function(){
+		if (this.id) {
+			history.replaceState({}, '', '#' + this.id);
```

Comment:
Nakolko by bolo zlozite, aby sa hash zmenil iba pri **expanduti** otazky, nie pri jej zatvoreni.

Reply:
Rozmyslal som nad tym aj predtym, location hash menim pri `show.bs.collapse` evente. Ale nie je to stopercentne pri zlozitejsich pripadoch. Ak na `data-target` selectne viac ako jeden element.
Priklad: **A** `data-hash="a"`, **B** `data-hash="b"`. Pri rozkliknuti **A** sa rozbali aj **B** a url sa zmeni na `www.example.com/#b`. 
Preto som pouzil event na click elementu. Ale ako si povedal vyssie ze zlozitejsie pripady by nemali nastat, tak to bude fungovat.

Reply:
Chapem. Hadam mozeme predpokladat, ze ak budeme niekedy potrebovat rozbalit naraz viac elementov, tak iba jeden z nich bude mat definovany atribut `data-hash`.

Reply:
Btw, nevedeli by sme v onclick handleri zobrat `$(this).data('target')` a pozriet, ci tento target element je collapsnuty? Teda ci ma class `collapse` alebo `in`? A podla toho sa rozhodnut, ci zavolame `history.replaceState(...)`?

Reply:
ono by to praveze zvladlo aj zlozitejsie pripady, a to tak ze pre **B** `data-hash="a"`.

Reply:
:ok: je to prerobene na onclick